### DISCUSSION TITLE
Data structure for galaxy cluster data

### DISCUSSION THREAD

**Discussion by bnord** [deepskies/DeepWiki#31] [Ideas]
How should we design our data structure for haloes and clusters? This applies to SZ, Optical, and X-ray.

Some options that I can think of:
1. Halo class that has properties (attributes) of both the DM halo and the cluster.
2. A set of lists: e.g., halo ID, halo mass, SZ signal

**Comment by kbanker** [comment]
Quick clarification, is this about use within the sim/in code, or for the outputs in h5?

**Reply by bnord** [reply]
both

**Reply by bnord** [reply]
I could imagine storing a list of halo objects in the h5. I think h5 permits that, but I'm not sure.

**Reply by simone-shev** [reply]
h5 doesn't actually permit that but there are [workarounds](https://stackoverflow.com/questions/48294013/how-to-store-my-own-class-object-into-hdf5)
